src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -76,4 +237,61 @@ export class RetireJsEngine implements RuleEngine {
 	public async isEnabled(): Promise<boolean> {
 		return await this.config.isEngineEnabled(RetireJsEngine.ENGINE_ENUM);
 	}
+
+	private async createPathAlias(tmpParent: string, basePath: string): Promise<string> {
+		// We want to make sure each duplicate file's path is unique, to avoid collisions. If we haven't already created
+		// a unique alias for this path's directory, we need to do that now.
+		const baseDir = path.dirname(basePath);
+		if (!this.aliasesByOriginalPath.has(baseDir)) {
+			const aliasDir = path.join(tmpParent, `TMP_${RetireJsEngine.NEXT_ALIAS_IDX++}`);
+			this.aliasesByOriginalPath.set(baseDir, aliasDir);
+			this.originalPathsByAlias.set(aliasDir, baseDir);
+			return new Promise((res, rej) => {
+				// Create a directory beneath the temporary parent directory with the name of the alias.
+				fs.mkdir(aliasDir, (err) => {
+					if (err) {
+						rej(err);
+					} else {
+						res(aliasDir);
+					}
+				});
+			});
+		}
+		return this.aliasesByOriginalPath.get(baseDir);
+	}
+
+
+	protected async createTmpDirWithDuplicatedTargets(targets: RuleTarget[]): Promise<string> {
+		// Create a temporary parent directory into which we'll transplant all of our target files.
+		const tmpParent: string = await new FileHandler().tmpDirWithCleanup();
+		const fileCopyPromises: Promise<void>[] = [];
+
+		// We want to duplicate all of the targeted files into our temporary directory.
+		for (const target of targets) {
+			for (const p of target.paths) {
+				// Create a unique alias directory into which we should duplicate all of these files.
+				// We'll use an `await` for this line because this operation needs to be atomic.
+				const pathAlias: string = await this.createPathAlias(tmpParent, p);
+
+				// Once we've got a path, ew can copy the original target file to the child directory, thereby preserving
+				// path uniqueness. No race condition exists, so no need for an `await`.
+				fileCopyPromises.push(new Promise((res, rej) => {
+					fs.copyFile(p, path.join(pathAlias, path.basename(p)), (err) => {
```

Comment:
Have you considered copying the soft link of the file instead of the full file? That way, if the machine is running low on usage memory, we won't be adding to the load.

Reply:
What's the soft link?

Reply:
@jfeingold35 Symbolic link. That way, it wouldn't hold a copy of the contents, but rather a pointer to the original file.
https://www.geeksforgeeks.org/node-js-fs-symlink-function/

Reply:
I'll take a look at this and see whether RetireJS can handle that. Good idea.